src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -250,29 +250,43 @@ public void UpdateTrackedDocument(DocumentUri uri, SourceText newSourceText)
         // Find the matching document from the LSP solutions.
         foreach (var (workspace, lspSolution, isForked) in lspSolutions)
         {
-            var document = await lspSolution.GetTextDocumentAsync(textDocumentIdentifier, cancellationToken).ConfigureAwait(false);
-            if (document != null)
+            var documents = await lspSolution.GetTextDocumentsAsync(textDocumentIdentifier.DocumentUri, cancellationToken).ConfigureAwait(false);
```

Comment:
I am curious about the decision to the check in this method, is there a reason the other callers would not want to "prioritize" matching non-misc over misc files?

Reply:
That's a good question. This logic depends on knowing that there's a misc files provider service, so we'd have to fetch that in the extension method. @dibarbet any thoughts?